main.py

```
@@ -0,0 +1,159 @@
+    
+    mined_block_2 = mine_block(block_2)
+    chain.add_block(mined_block_2)
+    
+    # Apply transactions
+    for tx in mined_block_2.transactions:
+        state.apply_transaction(tx)
+
+    # ------------------------------------------------------------------
+    # FINAL CHECKS
+    # ------------------------------------------------------------------
+    print(f"\n[7] Final State Check:")
```

Ruff (0.15.0)

[error] f-string without any placeholders

Remove extraneous `f` prefix

(F541)